static/js/components/FollowupRequestForm.jsx

```diff
@@ -0,0 +1,260 @@
+import React, { useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import * as Actions from '../ducks/source';
+
+
+const FollowupRequestForm = ({ source }) => {
+  const dispatch = useDispatch();
+  const { instrumentList, instrumentObsParams } = useSelector((state) => state.instruments);
+  const obsParams = instrumentObsParams; // Shorten to reduce line length below
+
+  const instIDToName = {};
+  instrumentList.forEach((instrumentObj) => {
+    instIDToName[instrumentObj.id] = instrumentObj.name;
+  });
+
+  const initialFormState = {
+    source_id: source.id,
+    instrument_id: "",
+    instrument_name: "",
+    start_date: "",
+    end_date: "",
+    filters: [],
+    exposure_time: null,
+    priority: ""
+  };
+  const [formState, setFormState] = useState({ ...initialFormState });
+
+  const handleInputChange = (e) => {
+    let newState = { ...formState };
+    newState.filters = [...formState.filters];
+
+    if (e.target.name.startsWith("filterCheckbox_")) {
+      const filter = e.target.name.split("filterCheckbox_")[1];
+      if (e.target.checked) {
+        newState.filters.push(filter);
+      } else {
+        newState.filters.splice(newState.filters.indexOf(filter), 1);
+      }
+    } else {
+      newState[e.target.name] = e.target.type === 'checkbox' ?
+        e.target.checked : e.target.value;
+
+      if (e.target.name === "instrument_id") {
+        // Reset form state using newly selected instrument ID & name
+        newState = {
+          ...initialFormState,
+          instrument_id: e.target.value,
+          instrument_name: instIDToName[e.target.value],
+          editable: obsParams[instIDToName[e.target.value]].requestsEditable
+        };
+      }
+    }
+    setFormState({
+      ...formState,
+      ...newState
+    });
+  };
+
+  const handleSubmit = () => {
```

Comment:
here's where you can validate the dates on the client side (e.g. https://www.telerik.com/kendo-react-ui/components/dateinputs/datepicker/forms/#toc-setting-validation-requirements)

Reply:
How does this look? https://reactdatepicker.com/

Reply:
Yes, good. For a picker this is fine. But does it also validate on the client side?

Reply:
It doesn't allow for invalid dates - if you enter some invalid text, upon element blur, it reverts back to the last valid input.

Reply:
Done.